### DISCUSSION TITLE
Is the user conda env really necessary?

### DISCUSSION THREAD

**Discussion by rarensu** [oobabooga/text-generation-webui#5967] [General]
Howdy, 
I want to try to deploy this on a shared cluster with a Lustre network filesystem. We have **file count** quotas per user, due to the way that Lustre filesystem works. If each user has their own conda env, they will quickly run out of files in their quota. All the files from the base env that the user env requires get linked individually. That's too many files. This isn't a problem with vanilla virtualenvs, where an  variable points to the base site packages _directory_ rather than file by file. So for vanilla virtualenvs, every user can have their own and it's not a problem for their file quota. I want to know if it's reasonable to try to switch from conda env to virtualenv. Otherwise, I seem to be stuck with forcing all the users to share a single conda env. I don't really love the single, shared env strategy because then the users can't update the requirements if they want to try out some extensions, for example. 
Richard

**Comment by strawberrymelonpanda** [comment]
Conda / Mamba / etc are just env managers.  Venv should work just fine if you're willing to change the various launcher scripts.

**Comment by rarensu** [comment]
I was able to get it to run out of a regular venv, and it seems to work correctly. I did have to modify a few sections of code, mostly to remove some conda sanity checks. I don't think it's worth a pull request, but I wanted to at least let others know that it is very achievable.
